File: src/main/java/com/saybetter/global/utils/SecurityUtil.java
package com.saybetter.global.utils;

import java.util.Collection;

import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

import lombok.NoArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@NoArgsConstructor
public class SecurityUtil {

	public void clearSecurityContext() {
		SecurityContextHolder.clearContext();
	}

	public String getCurrentUserEmail() {
		UsernamePasswordAuthenticationToken authentication =
				(UsernamePasswordAuthenticationToken)SecurityContextHolder.getContext().getAuthentication();
		return authentication.getName();
	}

	public Collection<GrantedAuthority> getUserAuthorities() {
		UsernamePasswordAuthenticationToken authentication =
				(UsernamePasswordAuthenticationToken)SecurityContextHolder.getContext().getAuthentication();
		return authentication.getAuthorities();
	}

}

File: core/core-api/src/main/kotlin/io/say/better/global/utils/SecurityUtil.kt
package io.say.better.global.utils

import io.say.better.global.config.logger.logger
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class SecurityUtil constructor() {
    private val log = logger()

    fun clearSecurityContext() {
        SecurityContextHolder.clearContext()
    }

    val currentUserEmail: String
        get() {
            val authentication =
                SecurityContextHolder.getContext().authentication as UsernamePasswordAuthenticationToken
            return authentication.name
        }

    val userAuthorities: Collection<GrantedAuthority>
        get() {
            val authentication =
                SecurityContextHolder.getContext().authentication as UsernamePasswordAuthenticationToken
            return authentication.authorities
        }
}
